### PR TITLE
fix : Verify the credit card number using Luhn's Algorithm while adding Card

### LINKED ISSUE #420: Limit the digits of Credit Card Number while adding a new card to Saved Cards.
**Summary:**

There should be a limit of digits while entering a Credit Card Number (In add card dialoge box)  to Saved Cards list and a validation check should be present to check the entered Credit card number is not less than 16 digits.

**Steps to reproduce:**

Home --> SavedCards --> click on Add Card button.

**Expected behavior:**

There shoud be a limit of digits while entering Credit/Debit Card Number and validation checker of length.

**Observed behavior:**

There is no limit of digits and no validation checker.

**Device and Android version:**

Device: Redmi 4
Version: 7.1.2

**Screenshots:**

### PR DESCRIPTION
**Pull request by RajaVamsi11**
Fixes #420 

**Summary**
Added validation checker and limited the digits length of Credit Card Number in Saved Cards.
screenshot_2019-03-06-18-17-12-189_org mifos mobilewallet mifospay

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by naman14** [comment]
@RajaVamsi11 
Why not use the credit card number verification method already present in `CardsPresenter`?
https://github.com/openMF/mobile-wallet/blob/a33c849d69fbd3e8f3d2ae2d9d18d64a130e1d74/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/savedcards/presenter/CardsPresenter.java#L195

Also, credit cards can have as many as 19 digits, there's no need to limit the number of digits. Rather use the credit card verification algorithm mentioned above.

**Comment by RajaVamsi11** [comment]
@naman14 
   
        int u = 2;
        if (u - 2 == 0) {
            return true; // for backend testing. remove after testing.
        }
According to the above code in the ValidateCreditCardNumberMethod() it always return true. It must be removed. Can I remove it as it says it is for backend testing, remove after testing.

**Comment by RajaVamsi11** [comment]
@naman14 Updated according to the changes requested. Can you please review once?

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/savedcards/ui/AddCardDialog.java [hunk 1]
```diff
@@ -147,18 +147,22 @@ public void onAddClicked() {
...
+                card.setId(editCard.getId());
+                mCardsPresenter.editCard(card);
+            } else {
+                mCardsPresenter.addCard(card);
+            }
+
+            dismiss();
         } else {
-            mCardsPresenter.addCard(card);
+            Toaster.showToast(getContext(), getString(R.string.enter_valid_card_number));
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/savedcards/ui/AddCardDialog.java]: @RajaVamsi11 Use Snackbar instead.

**Review by luckyman20 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**